test/jdk/java/awt/Menu/OnFlyRepaintMenuTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2004, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 5024051
+ * @summary Tests if menu is repainted in enabling/disabling it and
+ *          changing its label while it is visible, either on MenuBar
+ *          or in other Menu. Menu items are covered as well
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual OnFlyRepaintMenuTest
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Button;
+import java.awt.CheckboxMenuItem;
+import java.awt.Frame;
+import java.awt.Menu;
+import java.awt.MenuBar;
+import java.awt.MenuItem;
+
+public class OnFlyRepaintMenuTest {
+    static boolean menuEnabled = true;
+
+    public static void main(String[] args) throws Exception {
+        String INSTRUCTIONS = """
+                1. Click the button 'Change state' and wait for 5 secs.
+                2. If menu is repainted correctly after its setLabel()
+                   and setEnabled() methods called test PASSED, else FAILED.
```

Comment:
How to know that correct menu is repainted ?

You may add the initial menu details... like menu labels, menu state, menu checked state etc.
And then after clicks... what change is expected ?

This will give tester an info to compare against the test instructions.

Reply:
repainted correctly after setLabel and setEnabled is evident that its w.r.t to label and enable/disable.

Reply:
In that case, there is no info about `setDisable` method in test instruction.